gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java

```diff
@@ -133,101 +129,11 @@ private void rebuildForm(Form form) {
-        break;
-    }
-  }
-
-  private void addTextField(Field field) {
-    TextInputFieldBinding binding =
-        TextInputFieldBinding.inflate(getLayoutInflater(), formLayout, false);
-    binding.setViewModel(viewModel);
-    binding.setLifecycleOwner(this);
-    binding.setField(field);
-    formLayout.addView(binding.getRoot());
-    assignGeneratedId(binding.getRoot().findViewById(R.id.text_input_edit_text));
-  }
-
-  private void addMultipleChoiceField(Field field) {
-    MultipleChoiceInputFieldBinding binding =
-        MultipleChoiceInputFieldBinding.inflate(getLayoutInflater(), formLayout, false);
-    binding.setFragment(this);
-    binding.setViewModel(viewModel);
-    binding.setLifecycleOwner(this);
-    binding.setField(field);
-    formLayout.addView(binding.getRoot());
-    assignGeneratedId(binding.getRoot().findViewById(R.id.multiple_choice_input_edit_text));
-  }
-
-  private void addPhotoField(Field field) {
-    PhotoInputFieldBinding binding =
-        PhotoInputFieldBinding.inflate(getLayoutInflater(), formLayout, false);
-    binding.setLifecycleOwner(this);
-    binding.setField(field);
-    binding.setFragment(this);
-
-    PhotoFieldViewModel photoFieldViewModel = viewModelFactory.create(PhotoFieldViewModel.class);
-    photoFieldViewModel.init(field, viewModel.getResponses());
-    binding.setViewModel(photoFieldViewModel);
-
-    formLayout.addView(binding.getRoot());
-  }
-
-  public void onShowDialog(Field field) {
-    Cardinality cardinality = field.getMultipleChoice().getCardinality();
-    Optional<Response> currentResponse = viewModel.getResponse(field.getId());
-    switch (cardinality) {
-      case SELECT_MULTIPLE:
-        multiSelectDialogFactory
-            .create(field, currentResponse, r -> viewModel.onResponseChanged(field, r))
-            .show();
-        break;
-      case SELECT_ONE:
-        singleSelectDialogFactory
-            .create(field, currentResponse, r -> viewModel.onResponseChanged(field, r))
-            .show();
-        break;
-      default:
-        Timber.e("Unknown cardinality: %s", cardinality);
-        break;
-    }
-  }
-
-  public void onShowPhotoSelectorDialog(Field field) {
-    if (addPhotoBottomSheetBinding == null) {
-      addPhotoBottomSheetBinding = EditObservationBottomSheetBinding.inflate(getLayoutInflater());
-      addPhotoBottomSheetBinding.setViewModel(viewModel);
-    }
-    addPhotoBottomSheetBinding.setField(field);
-
-    if (bottomSheetDialog == null) {
-      bottomSheetDialog = new BottomSheetDialog(getContext());
-      bottomSheetDialog.setContentView(addPhotoBottomSheetBinding.getRoot());
-    }
-
-    if (!bottomSheetDialog.isShowing()) {
-      bottomSheetDialog.show();
-    }
-  }
-
   @Override
   public void onPause() {
-    if (bottomSheetDialog != null && bottomSheetDialog.isShowing()) {
-      bottomSheetDialog.dismiss();
+    for (FieldView fieldView : fieldViews) {
+      fieldView.onPause();
```

Comment:
Doesn't pause get propagated to child components automatically? And if not, wouldn't we also need to propagate other events like onStart()?

Reply:
It wasn't needed because we are re-creating the views and only VM is retained. `onPause()` was needed because I had to handle state of AlertDialog when app is paused.